service/policy/db/resource_mapping.go

```diff
@@ -2,18 +2,171 @@ package db
+	if err != nil {
+		return nil, err
+	}
+
+	resourceMappingGroups := make([]*policy.ResourceMappingGroup, len(list))
+
+	for i, rmGroup := range list {
+		resourceMappingGroups[i] = &policy.ResourceMappingGroup{
+			Id:          rmGroup.ID,
+			NamespaceId: rmGroup.NamespaceID,
+			Name:        rmGroup.Name,
+		}
+	}
+
+	return resourceMappingGroups, nil
+}
+
+func (c PolicyDBClient) ListResourceMappingGroupsByFqns(ctx context.Context, fqns []string) (map[string]*policy.ResourceMappingGroup, error) {
+	if (fqns == nil) || (len(fqns) == 0) {
+		return nil, fmt.Errorf("FQN list provided is empty")
+	}
+
+	resp := make(map[string]*policy.ResourceMappingGroup)
+	resultCount := 0
+
+	for _, fqn := range fqns {
+		parsedFqn, err := util.ParseResourceMappingGroupFqn(fqn)
+		if err != nil {
+			// FQN is not valid so ignore it and move to next
+			// invalid FQNs will not be included in the response
+			continue
+		}
+
+		rmGroup, err := c.Queries.GetResourceMappingGroupByFqn(ctx, GetResourceMappingGroupByFqnParams{
+			NamespaceName: parsedFqn.Namespace,
+			GroupName:     parsedFqn.GroupName,
+		})
+		if err != nil {
+			dbErr := db.WrapIfKnownInvalidQueryErr(err)
+			if errors.Is(dbErr, db.ErrNotFound) {
+				// FQN is valid but not in db, so indicate it was not found with nil value
+				resp[fqn] = nil
+				continue
+			}
+
+			return nil, dbErr
+		}
+
+		resultCount++
+
+		resp[fqn] = &policy.ResourceMappingGroup{
+			Id:          rmGroup.ID,
+			NamespaceId: rmGroup.NamespaceID,
+			Name:        rmGroup.Name,
+		}
+	}
+
+	if resultCount == 0 {
+		// should return an error if none of the FQNs are found
+		return nil, db.ErrNotFound
+	}
+
+	return resp, nil
+}
+
+func (c PolicyDBClient) GetResourceMappingGroup(ctx context.Context, id string) (*policy.ResourceMappingGroup, error) {
+	rmGroup, err := c.Queries.GetResourceMappingGroup(ctx, id)
+	if err != nil {
+		return nil, db.WrapIfKnownInvalidQueryErr(err)
+	}
+
+	return &policy.ResourceMappingGroup{
+		Id:          rmGroup.ID,
+		NamespaceId: rmGroup.NamespaceID,
+		Name:        rmGroup.Name,
+	}, nil
+}
+
+func (c PolicyDBClient) CreateResourceMappingGroup(ctx context.Context, r *resourcemapping.CreateResourceMappingGroupRequest) (*policy.ResourceMappingGroup, error) {
+	createdID, err := c.Queries.CreateResourceMappingGroup(ctx, CreateResourceMappingGroupParams{
+		NamespaceID: r.GetNamespaceId(),
+		Name:        r.GetName(),
+	})
+	if err != nil {
+		return nil, db.WrapIfKnownInvalidQueryErr(err)
+	}
+
+	return &policy.ResourceMappingGroup{
+		Id: createdID,
+	}, nil
+}
+
+func (c PolicyDBClient) UpdateResourceMappingGroup(ctx context.Context, id string, r *resourcemapping.UpdateResourceMappingGroupRequest) (*policy.ResourceMappingGroup, error) {
+	uuidNamespaceID, err := uuid.Parse(r.GetNamespaceId())
+	pgNamespaceID := pgtype.UUID{
+		Bytes: [16]byte(uuidNamespaceID),
+		Valid: err == nil,
+	}
+
+	name := r.GetName()
+	pgName := pgtype.Text{
+		String: name,
+		Valid:  name != "",
+	}
```

Comment:
I actually almost went down the NULLIF road, but yes you have to use `uuid.Parse()` to be able to set the `pgtype.UUID.Bytes` variable because they don't provide any sort of convenience function for passing in a string.  I could try NULLIF.

Reply:
On second thought, let's move this to a future concern: https://github.com/opentdf/platform/issues/1378